tests/Doctrine/Tests/Common/Proxy/ProxyGeneratorTest.php

```diff
@@ -229,6 +229,92 @@ public function testClassWithReturnTypesOnProxiedMethods()
+        }
+
+        if (!class_exists('Doctrine\Tests\Common\ProxyProxy\__CG__\NullableTypeHintsClass', false)) {
+            $className = NullableTypeHintsClass::class;
+            $metadata = $this->createClassMetadata($className, ['id']);
+
+            $proxyGenerator = new ProxyGenerator(__DIR__ . '/generated', __NAMESPACE__ . 'Proxy', true);
+            $this->generateAndRequire($proxyGenerator, $metadata);
+        }
+
+        $classCode = file_get_contents(__DIR__ . '/generated/__CG__DoctrineTestsCommonProxyNullableTypeHintsClass.php');
+
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintInt(?int $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintObject(?\stdClass $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintSelf(?\\' . $className . ' $param)'));
+        $this->assertEquals(1, substr_count($classCode, 'function nullableTypeHintWithDefaultNull(?int $param = NULL)'));
+    }
+
+    public function testClassWithNullableReturnTypesOnProxiedMethods()
+    {
+        if (PHP_VERSION_ID < 70100) {
+            $this->markTestSkipped('Nullable method return types are only supported in PHP >= 7.1.0.');
+        }
+
+        $className = NullableTypeHintsClass::class;
+        if (!class_exists('Doctrine\Tests\Common\ProxyProxy\__CG__\NullableTypeHintsClass', false)) {
+            $metadata = $this->createClassMetadata($className, ['id']);
+
+            $proxyGenerator = new ProxyGenerator(__DIR__ . '/generated', __NAMESPACE__ . 'Proxy', true);
+            $this->generateAndRequire($proxyGenerator, $metadata);
+        }
+
+        $classCode = file_get_contents(__DIR__ . '/generated/__CG__DoctrineTestsCommonProxyNullableTypeHintsClass.php');
+
+        $this->assertEquals(1, substr_count($classCode, 'function returnsNullableInt(): ?int'));
+        $this->assertEquals(1, substr_count($classCode, 'function returnsNullableObject(): ?\stdClass'));
+        $this->assertEquals(1, substr_count($classCode, 'function returnsNullableSelf(): ?\\' . $className));
+    }
+
+    public function testClassWithVoidReturnType()
+    {
+        if (PHP_VERSION_ID < 70100) {
+            $this->markTestSkipped('Void method return types are only supported in PHP >= 7.1.0.');
+        }
+
+        $className = VoidReturnTypeClass::class;
+        if (!class_exists('Doctrine\Tests\Common\ProxyProxy\__CG__\VoidReturnTypeClass', false)) {
+            $metadata = $this->createClassMetadata($className, ['id']);
+
+            $proxyGenerator = new ProxyGenerator(__DIR__ . '/generated', __NAMESPACE__ . 'Proxy', true);
+            $this->generateAndRequire($proxyGenerator, $metadata);
+        }
+
+        $classCode = file_get_contents(__DIR__ . '/generated/__CG__DoctrineTestsCommonProxyVoidReturnTypeClass.php');
+
+        $this->assertEquals(1, substr_count($classCode, 'function returnsVoid(): void'));
```

Comment:
What exactly do you mean? Since we can't put any return into the method that is to be proxied, just testing that such method returns NULL, i.e. `assertNull($proxy->returnsVoid())`?